File: DiagnosticService/Transport/RetroQuery.cs
using System;
using ProtoBuf;

namespace Diagnostics.Service.Common.Transport;

[ProtoContract]
public class RetroQuery
{
    public int SearchId { get; set; }
    public int MaxRecords { get; set; }
    public int MinLevel { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public string? Machine { get; set; }
    public string? Process { get; set; }
    public string? User { get; set; }
    public string? Message { get; set; }

    public RetroQuery Clone()
    {
        return (RetroQuery)MemberwiseClone();
    }
}

File: angular-app/src/app/members/diagnostics/Model/RetroQuery.ts
import {DiagnosticMsg} from './DiagnosticMsg';

export class RetroQuery {
    searchId = 0;
    maxRecords = 1000;
    minLevel = 0;
    startDate?: string;
    endDate?: string;
    machine?: string;
    process?: string;
    user?: string;
    message?: string;
}

export class RetroSearchResult {
    searchId = 0;
    info = '';
    results: DiagnosticMsg[] = [];
}
